### PR TITLE
ci(dev_env): Test development environment on Big Sur and Ubuntu

### PR DESCRIPTION
**Pull request by armenzg**
We have users out in the wild using `brew` on Linux, thus, we will serve our community
by testing the development environment there as well.

I'm also adding Big Sur back into the matrix (even if it's slower to execute) in order
to make sure we don't regress it when we make changes to the shared Brewfile.

### CODE REVIEW

File: Brewfile
Comment (armenzg) [inline Brewfile:14]: This package is only necessary for Ubuntu (as per #23002)

File: .github/workflows/development-environment.yml
Comment (billyvg) [inline .github/workflows/development-environment.yml:19]: Can we add a comment to clarify what the different os tags mean? at least for macos

**Review by billyvg — APPROVED** [review]
LGTM - let's wait for a response from #23002 before merging

File: .github/workflows/development-environment.yml
Comment (armenzg) [inline .github/workflows/development-environment.yml:14]: This change and the one above helps to fit the full string name in the dropdown since it now contains the OS name.

Before:
image

After:
image

File: .github/workflows/development-environment.yml
Comment (armenzg) [inline .github/workflows/development-environment.yml:24]: This permits all three platforms to complete to the end.

File: .github/workflows/development-environment.yml
Comment (armenzg) [inline .github/workflows/development-environment.yml:39]: I want to see if this issue happens anymore now that we try two times to execute `brew bundle`.

**Comment by armenzg** [comment]
Hi, @billyvg I did few more changes to make this more stable.

File: .github/workflows/development-environment.yml
Comment (armenzg) [inline .github/workflows/development-environment.yml:23]: Before I had `ubuntu-latest`, however, that can sometimes be 18.04 and sometimes 20.4. For now, let's pin it to one since 20.04 fails.

**Review by billyvg — APPROVED** [review]

**Pull request merged**
